### PR TITLE
Lb fluid refactor

### PR DESCRIPTION
**Pull request by fweik**
Description of changes:
 - Replaced lbfluid by multi_grid

This is just an intermediate solution, because I didn't wanted to
change the indexing before the release.

### CODE REVIEW

**Comment by fweik** [comment]
Argh this has a 20% performance regression, because multi_array does not support move semantics and the swap will make copies. Will fix.

File: src/core/virtual_sites/lb_inertialess_tracers.cpp [hunk 1]
```diff
@@ -5,22 +5,23 @@
```
Comment (KaiSzuttor) [inline src/core/virtual_sites/lb_inertialess_tracers.cpp:5]: did you change anything in that file besides format?

**Review by KaiSzuttor — APPROVED** [review]
LGTM

File: src/core/virtual_sites/lb_inertialess_tracers.cpp (hunk 1 above)
Comment (fweik) [reply src/core/virtual_sites/lb_inertialess_tracers.cpp:5]: No I don't think so. It's formated because there where intermediate changes to it.

**Comment by fweik** [comment]
@KaiSzuttor ready for review
Performance is now unchanged, according to this benchmark:
```python
from __future__ import print_function

import time
import numpy as np

import espressomd
import espressomd.lb

cells_per_side=16
n_cells = cells_per_side**3
agrid = 1.
box_l = cells_per_side*agrid

system = espressomd.System(box_l=[box_l]*3)

system.time_step = 0.01
system.cell_system.skin = 0.1
lb_params = {'agrid': agrid, 'fric': 1, 'dens': 1, 'visc': 1, 'tau': 0.01,
             'ext_force_density': [0, 0, -1.0 / (box_l**3)]}
lbf = espressomd.lb.LBFluid(**lb_params)
system.actors.add(lbf)

steps=1000
n_nodes = system.cell_system.get_state()['n_nodes']

while True:
    tick = time.time()
    system.integrator.run(steps=steps)
    tock = time.time()
    dt = tock - tick
    l = steps*n_cells / dt / 1000.

    print("{} kLU/s, {} kLU/s/node".format(l, l/n_nodes))
```

**Comment by KaiSzuttor** [comment]
```/home/espresso/travis/src/core/lb.cpp:2550:44: error: invalid initialization of non-const reference of type 'boost::multi_array<double, 2ul>&' from an rvalue of type 'boost::multi_array<double, 2ul>'```

**Comment by KaiSzuttor** [comment]
I just measured with your sample script...
single core: approx. 1.3 times faster
four cores: approx. 2.6 times faster
compared to current python branch.

LGTM

**Review by KaiSzuttor — APPROVED** [review]

**Comment by fweik** [comment]
It's just faster because the particle coupling is now skipped when there are no particles, which steps over the halo comm. So the speedup isn't really relevant in practice.

**Comment by KaiSzuttor** [comment]
just checked with added particle, then they are very similar

**Pull request merged**